### PR TITLE
feat: receipt upload ui

### PR DESCRIPTION
**Pull request by tomaszantas**
## Description

Add new UI for receipt upload in Bank Transactions

image

### CODE REVIEW

File: src/components/ExpandedBankTransactionRow/ExpandedBankTransactionRow.tsx [hunk 1]
```diff
@@ -617,38 +674,30 @@ export const ExpandedBankTransactionRow = forwardRef<SaveHandle, Props>(
...
-                          bankTransactionId: bankTransaction.id,
-                          file: file,
-                          documentType: 'RECEIPT',
-                        })
-                        await fetchDocuments()
-                      }}
-                      text='Upload receipt'
+                      secondary
+                      onUpload={onReceiptUpload}
+                      text='Add next receipt'
```
Comment (AndrewAcomb) [inline src/components/ExpandedBankTransactionRow/ExpandedBankTransactionRow.tsx:698]:
```suggestion
                      text='Upload receipt'
```

File: src/components/ExpandedBankTransactionRow/ExpandedBankTransactionRow.tsx [hunk 2]
```diff
@@ -441,14 +477,101 @@ export const ExpandedBankTransactionRow = forwardRef<SaveHandle, Props>(
 
     useEffect(() => {
       const loadDocumentsAndMetadata = async () => {
-        if (showDescriptions) await fetchMetadata()
-        if (showReceiptUploads) await fetchDocuments()
+        if (showDescriptions && bankTransaction?.metadata.memo)
+          setMemoText(bankTransaction?.metadata.memo)
```
Comment (AndrewAcomb) [inline src/components/ExpandedBankTransactionRow/ExpandedBankTransactionRow.tsx:481]: Made this change so it doesn't separately call the metadata endpoint for each transaction on initial load, instead using the metadata field on the bank transaction object.

File: src/components/BankTransactionsLoader/BankTransactionsLoader.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ import { SkeletonTableLoader } from '../SkeletonTableLoader'
 
 export const BankTransactionsLoader = ({
   isLoading = true,
-  showTooltips
+  showTooltips,
```
Comment (AndrewAcomb) [inline src/components/BankTransactionsLoader/BankTransactionsLoader.tsx:11]: This would certainly be in a different PR, but I am curious what you think about potentially using [Jotai](https://jotai.org/) at some point, since I did a lot of prop drilling for the tooltips / receipt upload props.

**Comment by AndrewAcomb** [comment]
https://github.com/user-attachments/assets/238f773f-7cc8-44ee-910e-f10c4415942e

When there is an error state, you can't click on any of the actions since that requires you to hover on the receipt (which the actions do not overlap) ^ see video

File: src/components/BankTransactionsLoader/BankTransactionsLoader.tsx (hunk 3 above)
Comment (tomaszantas) [reply src/components/BankTransactionsLoader/BankTransactionsLoader.tsx:11]: Definitely there are good use cases for Jotai. The only two minor concerns I have are:
- NPM package weights a bit, ~427kB... but still worth to use imho
- we will need to be careful with mixing atoms with specific local states and data later sent to API

**Comment by tomaszantas** [comment]
@AndrewAcomb here is a fix for the issue from your video: #362 .
I put this into new PR in case if you have some local changes for this branch.
Feel free to merge if all looks good.

**Review by AndrewAcomb — COMMENTED** [review]
Approving along with https://github.com/Layer-Fi/layer-react/pull/362 .

File: src/components/BankTransactionsLoader/BankTransactionsLoader.tsx (hunk 3 above)
Comment (AndrewAcomb) [reply src/components/BankTransactionsLoader/BankTransactionsLoader.tsx:11]: We had a brief discussion and agreed we should just move some things into context at this stage instead, but will revisit Jotai if we later have issues with too many re-renders (which we have not thus far).

**Review by AndrewAcomb — APPROVED** [review]
